apicast/src/provider.lua

```diff
@@ -385,8 +385,15 @@ function _M.access(service)
     ngx.var.cached_key = concat({service.id, params.app_id, params.app_key}, ':')
 
   elseif backend_version == 'oauth' then
-    ngx.var.access_token = parameters.access_token
-    params.access_token = parameters.access_token
+    if credentials.location == 'headers' then
```

Comment:
Yes. The other out modes support the same. But there is already uniform way to transfer credentials - `Authorization` header.

To minimize time to deliver this, we can just start checking Authorization header no matter what auth mode or credentials location. If other credentials are not found.
And later introduce UI option to enforce passing credentials in HTTP Authorization header.